### PR TITLE
chore: update to Rust 1.72

### PR DESCRIPTION
**Pull request by linbingquan**
As the title.

### CODE REVIEW

File: cli/napi/js_native_api.rs [hunk 1]
```diff
@@ -1631,7 +1631,7 @@ fn napi_delete_property(
 // TODO: properly implement ref counting stuff
 #[napi_sym::napi_sym]
 fn napi_delete_reference(env: *mut Env, _nref: napi_ref) -> napi_status {
-  let mut _env = &mut *(env as *mut Env);
```
Comment (mmastrac) [inline cli/napi/js_native_api.rs:1634]: I think these lines can all go away -- they should not have any effect.

Comment (linbingquan) [reply cli/napi/js_native_api.rs:1634]: > I think these lines can all go away -- they should not have any effect.

Ok, I remove the unused code.

File: cli/napi/js_native_api.rs [hunk 2]
```diff
@@ -1869,7 +1867,7 @@ fn napi_get_instance_data(
   env: *mut Env,
   result: *mut *mut c_void,
 ) -> napi_status {
-  let env = &mut *(env as *mut Env);
+  let env = &mut *env;
```
Comment (mmastrac) [inline cli/napi/js_native_api.rs:1870]: There's a few more. I think all instances of `let env = &mut *env;` are unused -- we can delete all the lines with this code.

**Comment by mmastrac** [comment]
Looks go to me to land for the next minor release.

File: cli/napi/js_native_api.rs (hunk 2 above)
Comment (linbingquan) [reply cli/napi/js_native_api.rs:1870]: > There's a few more. I think all instances of `let env = &mut *env;` are unused -- we can delete all the lines with this code.

I found `let env = &mut *env;` was used in next line, like:

```rs
#[napi_sym::napi_sym]
fn napi_get_instance_data(
  env: *mut Env,
  result: *mut *mut c_void,
) -> napi_status {
  let env = &mut *env;
  let shared = env.shared(); // This line was used `env` variable
  *result = shared.instance_data;
  napi_ok
}
```

It does not make sense in this `napi_get_instance_data` function.

I'm sorry, I don't know what you mean about "we can delete all the lines with this code". Could you tell me more about that.

**Comment by linbingquan** [comment]
@mmastrac I don't know why it can't pass the test in ubuntu.

https://github.com/denoland/deno/actions/runs/5971650008/job/16201009951?pr=20258

File: cli/napi/js_native_api.rs (hunk 2 above)
Comment (mmastrac) [reply cli/napi/js_native_api.rs:1868]: Ah sorry, yeah I see where this is used now.

**Comment by mmastrac** [comment]
@linbingquan Please try bumping the version of `async-trait` to `1.1.73` in Cargo.toml to fix that lint issue.

**Review by mmastrac — APPROVED** [review]

**Pull request merged**